styles/transition_loop.h

```diff
@@ -34,4 +34,61 @@ class TransitionLoopL {
 template<class T, class TRANSITION>
   using TransitionLoop = Layers<T, TransitionLoopL<TRANSITION>>;
 
+// Usage: TransitionLoopWhileL<LOOP_TRANSITION, END_TRANSITION, CONDITION>
+// LOOP_TRANSITION : TRANSITION
+// END_TRANSITION : TRANSITION
+// return value: COLOR
+//
+// Continuously runs LOOP_TRANSITION while CONDITION > 0
+// Makes more sense if TRANSITION is a TrConcat, as this will
+// transition to/from the intermediate steps in a loop.
+// If CONDITION <= 0, runs END_TRANSITION and stops.
+
+template<class LoopTr, class EndTr, class CONDITION>
+class TransitionLoopWhileL {
+public:
+  void run(BladeBase* blade) {
+    condition_.run(blade);
+    bool cond = condition_.calculate(blade);
+    if (!run_ || loop_tr_.done()) {
+       if (cond) {
+          run_ = true;
+          loop_tr_.begin();
+       }
+    }
+    if (run_) {
+       if (!end_ && !cond) {
+         end_ = true;
+         end_tr_.begin();
+       }
+    }
+    if (run_) {
+      loop_tr_.run(blade);
+      if (end_) end_tr_.run(blade);
+    }
+    if (end_ && end_tr_.done()) {
+      end_ = false;
+      run_ = false;
+    }
+  }
+  
+private:
+  bool run_ = false;
+  bool end_ = false;
+  LoopTr loop_tr_;
+  EndTr end_tr_;
+  PONUA SVFWrapper<CONDITION> condition_;
+
+public:
+auto getColor(int led) -> decltype(
+    MixColors(end_tr_.getColor(loop_tr_.getColor(RGBA_um_nod::Transparent(), RGBA_um_nod::Transparent(), led), RGBA_um_nod::Transparent(), led),
+        loop_tr_.getColor(loop_tr_.getColor(RGBA_um_nod::Transparent(), RGBA_um_nod::Transparent(), led), RGBA_um_nod::Transparent(), led), 1, 1)) {
+    decltype(MixColors(end_tr_.getColor(loop_tr_.getColor(RGBA_um_nod::Transparent(), RGBA_um_nod::Transparent(), led), RGBA_um_nod::Transparent(), led),
+        loop_tr_.getColor(loop_tr_.getColor(RGBA_um_nod::Transparent(), RGBA_um_nod::Transparent(), led), RGBA_um_nod::Transparent(), led), 1, 1)) ret = RGBA_um_nod::Transparent();
+    if (end_) ret = end_tr_.getColor(loop_tr_.getColor(RGBA_um_nod::Transparent(), RGBA_um_nod::Transparent(), led), RGBA_um_nod::Transparent(), led);
```

Comment:
line 88 and 89 needs to switch places.
The end transition needs to run "on top of" the loop, which means it needs to be after in the code.
Also, the first input to the end end transition should be "ret", not transparent.

the return type needs to be fixed in a similar manner too..

Reply:
"Also, the first input to the end end transition should be "ret", not transparent."

I didn't quite get this, how should it be?

Reply:
I just realized one more thing, if run_ and loop_ are both true, then this will run loop_tr_ twice, I think it should be like this:

```
if (run_) {
  ret = loop_tr_.getColor(RGBA_um_nod::Transparent(), RGBA_um_nod::Transparent(), led); 
  if (end_) ret = end_tr_.getColor(ret, RGBA_um_nod::Transparent(), led);
}
```